Compilation failure when minio-java is used as dependency as some API are no more available at build time

Due to folllowing change https://github.com/minio/minio-java/commit/8c7a5be3a42ae0e1f4da9269e97a30729b3cbf83#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7R54  in 8.5.0, dependencies like guava, okhttp and jackson are provided in runtime while there was in compile scope before.
Thus, some API like `MinioAsyncClient.builder().httpClient(httpClient)` no more compile.
```
/builds/.../s3/S3Storage.java:[33,15] package okhttp3 does not exist
[ERROR] /builds/.../s3/S3Storage.java:[47,23] cannot find symbol
  symbol:   class OkHttpClient
  location: class ....s3.S3Storage
```

I have found the following woraround in my maven project by adding the dependencyManagement part:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<artifactId>s3-storage</artifactId>
	<name>MyProject:: S3 Storage</name>

	<dependencies>
		<dependency>
			<groupId>io.minio</groupId>
			<artifactId>minio</artifactId>
                        <version>8.5.0</version>
		</dependency>
	</dependencies>
	
	<dependencyManagement>
		<dependencies>
			<dependency>
				<groupId>com.squareup.okhttp3</groupId>
				<artifactId>okhttp</artifactId>
				<scope>compile</scope>
			</dependency>
			<dependency>
				<groupId>com.fasterxml.jackson.core</groupId>
				<artifactId>jackson-databind</artifactId>
				<scope>compile</scope>
			</dependency>
			<dependency>
				<groupId>com.google.guava</groupId>
				<artifactId>guava</artifactId>
				<scope>compile</scope>
			</dependency>
		</dependencies>
	</dependencyManagement>
</project>
```

--- a/build.gradle
+++ b/build.gradle
@@ -41,7 +41,7 @@ allprojects {
 }
 
 subprojects {
-    apply plugin: 'java'
+    apply plugin: 'java-library'
     apply plugin: 'com.github.spotbugs'
     apply plugin: "com.diffplug.spotless"
 
@@ -51,15 +51,15 @@ subprojects {
     }
 
     dependencies {
-        implementation "com.carrotsearch.thirdparty:simple-xml-safe:2.7.1"
-        implementation "com.google.guava:guava:30.1.1-jre"
-        implementation "com.squareup.okhttp3:okhttp:4.10.0"
-        implementation "com.fasterxml.jackson.core:jackson-annotations:2.13.4"
-        implementation "com.fasterxml.jackson.core:jackson-core:2.13.4"
-        implementation "com.fasterxml.jackson.core:jackson-databind:2.13.4.2"
-        implementation "org.bouncycastle:bcprov-jdk15on:1.69"
-        implementation 'org.apache.commons:commons-compress:1.21'
-        implementation "org.xerial.snappy:snappy-java:1.1.8.4"
+        api "com.carrotsearch.thirdparty:simple-xml-safe:2.7.1"
+        api "com.google.guava:guava:30.1.1-jre"
+        api "com.squareup.okhttp3:okhttp:4.10.0"
+        api "com.fasterxml.jackson.core:jackson-annotations:2.13.4"
+        api "com.fasterxml.jackson.core:jackson-core:2.13.4"
+        api "com.fasterxml.jackson.core:jackson-databind:2.13.4.2"
+        api "org.bouncycastle:bcprov-jdk15on:1.69"
+        api 'org.apache.commons:commons-compress:1.21'
+        api "org.xerial.snappy:snappy-java:1.1.8.4"
         compileOnly "com.github.spotbugs:spotbugs-annotations:4.1.2"
 
         testImplementation "com.squareup.okhttp3:mockwebserver:4.8.1"
@@ -125,7 +125,7 @@ subprojects {
 }
 
 project(':api') {
-    apply plugin: 'java'
+    apply plugin: 'java-library'
     apply plugin: 'maven-publish'
     apply plugin: 'signing'
     apply plugin: 'com.github.johnrengelman.shadow'
@@ -232,7 +232,7 @@ project(':api') {
 }
 
 project(':adminapi') {
-    apply plugin: 'java'
+    apply plugin: 'java-library'
     apply plugin: 'maven-publish'
     apply plugin: 'signing'
     apply plugin: 'com.github.johnrengelman.shadow'
@@ -241,8 +241,8 @@ project(':adminapi') {
     archivesBaseName = 'minio-admin'
 
     dependencies {
-        implementation project(':api')
-        implementation "com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.13.4"
+        api project(':api')
+        api "com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.13.4"
         testImplementation project(':api')
     }
 

